### PR TITLE
feat(HQM-171): Sync NFT with indexer data

### PR DESCRIPTION
**Pull request by devkudasov**
Fixes #

## Proposed Changes

-
-
-

## Checklist

- [x] I have read the rules of [**contribution**](https://github.com/haqq-network/haqq-wallet/blob/main/docs/contribution.md#important-steps).

### CODE REVIEW

File: src/components/nft-item-details.tsx [hunk 1]
```diff
@@ -21,7 +21,7 @@ export interface NftItemDetailsProps {
 export const NftItemDetails = ({item, onPressSend}: NftItemDetailsProps) => {
   const [imageLayout, onImageLayout] = useLayout();
   const lastSalePrice = useMemo(
-    () => cleanNumber(parseInt(item.last_sale_price, 16) / WEI),
+    () => cleanNumber(parseInt(item.price, 16) / WEI),
```
Comment (vivalaakam) [inline src/components/nft-item-details.tsx:24]: use Balance

File: src/components/nft-viewer/nft-viewer-item-preview/nft-viewer-item-preview-large.tsx [hunk 2]
```diff
@@ -19,7 +19,7 @@ export const NftViewerItemPreviewLarge = ({
   const handlePress = useCallback(() => onPress?.(item), [onPress, item]);
   const [layout, onLayout] = useLayout();
   const lastSalePrice = useMemo(
-    () => cleanNumber(parseInt(item.last_sale_price, 16) / WEI),
+    () => cleanNumber(parseInt(item.price, 16) / WEI),
```
Comment (vivalaakam) [inline src/components/nft-viewer/nft-viewer-item-preview/nft-viewer-item-preview-large.tsx:22]: use Balance

File: src/components/nft-viewer/nft-viewer.tsx [hunk 3]
```diff
@@ -78,11 +60,11 @@ export const NftViewer = ({
   const [sortFieldName, setSortFieldName] =
     useState<keyof NftCollection>('created_at');
 
-  const sections = useMemo(
+  const sections: NftSection[] = useMemo(
     () =>
       data
-        ?.map?.(it => ({...it, data: [it]}) as SectionElement)
-        ?.sort?.(arraySortUtil(sortDirection, sortFieldName)),
+        .map(item => ({...item, data: [{data: item.data}]}) as NftSection)
```
Comment (ragozin-nikita) [inline src/components/nft-viewer/nft-viewer.tsx:66]: What is the reason of duplicating the `item.data` field? You can get it from the `item` object

File: src/components/nft-viewer/nft-viewer.tsx [hunk 4]
```diff
@@ -123,8 +105,8 @@ export const NftViewer = ({
   }, [animate, viewMode]);
 
   const onNftCollectionPress = useCallback(
-    (item: NftCollection) => {
-      navigation.navigate('nftDetails', {type: 'collection', item});
+    (collectionId: HaqqCosmosAddress) => {
+      navigation.navigate('nftDetails', {type: 'collection', collectionId});
```
Comment (ragozin-nikita) [inline src/components/nft-viewer/nft-viewer.tsx:109]:
```suggestion
      navigation.navigate(Stack.Page, {type: 'collection', collectionId});
```

File: src/components/nft-viewer/nft-viewer.tsx [hunk 5]
```diff
@@ -190,18 +144,16 @@ export const NftViewer = ({
...
       <First>
         {viewMode === NftViewerViewMode.collectionListWithItems && (
-          <SectionList
+          <NftViewerCollectionPreviewList
+            data={sections}
+            onItemPress={onNftItemPress}
+            onCollectionPress={onNftCollectionPress}
             scrollEnabled={scrollEnabled}
-            sections={sections}
-            keyExtractor={keyExtractor}
```
Comment (ragozin-nikita) [inline src/components/nft-viewer/nft-viewer.tsx:196]: Missing `keyExtractor`

File: src/contexts/app.ts [hunk 6]
```diff
@@ -548,6 +549,7 @@ class App extends AsyncEventEmitter {
 
     if (changed) {
       Token.fetchTokens();
```
Comment (ragozin-nikita) [inline src/contexts/app.ts:551]:
```suggestion
      Promise.all([Token.fetchTokens(), Nft.fetchNft()]);
```

File: src/models/nft.ts [hunk 7]
```diff
@@ -0,0 +1,208 @@
...
+
+  constructor(shouldSkipPersisting: boolean = false) {
+    makeAutoObservable(this);
+    if (!shouldSkipPersisting) {
+      makePersistable(this, {
+        name: this.constructor.name,
+        properties: [
+          // https://github.com/quarrant/mobx-persist-store/issues/97
+          // @ts-ignore
+          'nfts',
```
Comment (ragozin-nikita) [inline src/models/nft.ts:40]:
```suggestion
          'data',
```

File: src/models/nft.ts [hunk 8]
```diff
@@ -0,0 +1,208 @@
...
+
+        const nftItem: NftItem = {
+          id: `${contract.id}_${item.token_id}`,
+          contract: item.contract,
+          address: item.address,
+          name: item.name,
+          description: item.description,
+          created_at: item.created_at,
+          price: '100',
+          image: item.cached_url ?? 'https://i.ibb.co/9VGgYqf/10.jpg',
```
Comment (ragozin-nikita) [inline src/models/nft.ts:180]:
```suggestion
          image: item.cached_url ?? DEFAULT_NFT_IMAGE
```

File: src/widgets/nft-widget/index.tsx [hunk 9]
```diff
@@ -66,7 +54,7 @@ export const NftWidgetWrapper = observer(({size}: INftWidget) => {
     case NftWidgetSize.large:
       return (
         <ShadowCard disablePadding onPress={onPress} style={styles.wrapper}>
-          <NftCollectionInfoBanner data={nftCollections} />
+          <NftCollectionInfoBanner data={allNft} />
```
Comment (ragozin-nikita) [inline src/widgets/nft-widget/index.tsx:57]:
```suggestion
          <NftCollectionInfoBanner data={Nft.getAll()} />
```

**Review by ragozin-nikita — CHANGES_REQUESTED** [review]

File: src/components/nft-collection-details.tsx [hunk 10]
```diff
@@ -1,61 +1,62 @@
...
+          {/*TODO Description support for collections*/}
+          {/*<TrimmedText limit={100} t14 color={Color.textBase2}>*/}
+          {/*  {collection.description}*/}
+          {/*</TrimmedText>*/}
+          <Spacer height={16} />
+          <Text t12 i18n={I18N.nftDetailsYourItems} />
+          <Spacer height={8} />
+          <View>
+            <NftViewerItemPreviewList
+              scrollEnabled={false}
```
[the remark this answers is not in the document]
Comment (devkudasov) [reply src/components/nft-collection-details.tsx:49]: It is always false. I removed it.

[the remark this answers is not in the document]
Comment (devkudasov) [reply src/components/nft-collection-details.tsx:49]: We use this component in 3 places and scrollEnabled 2 times false and 1 time true

File: src/components/nft-viewer/nft-viewer-collection-preview/nft-viewer-collection-preview.tsx [hunk 11]
```diff
@@ -6,22 +6,22 @@ import {Color} from '@app/colors';
 import {createTheme} from '@app/helpers';
 import {useLayout} from '@app/hooks/use-layout';
 import {I18N} from '@app/i18n';
-import {NftCollection} from '@app/types';
+import {HaqqCosmosAddress, NftCollection} from '@app/types';
 import {addOpacityToColor} from '@app/utils';
 
-import {Text} from '../ui';
+import {Text} from '../../ui';
```
[the remark this answers is not in the document]
Comment (devkudasov) [reply src/components/nft-viewer/nft-viewer-collection-preview/nft-viewer-collection-preview.tsx:12]: Fixed

File: src/components/nft-viewer/nft-viewer.tsx (hunk 4 above)
Comment (devkudasov) [reply src/components/nft-viewer/nft-viewer.tsx]: Fixed

File: src/components/nft-viewer/nft-viewer.tsx (hunk 3 above)
Comment (devkudasov) [reply src/components/nft-viewer/nft-viewer.tsx]: This is for section list. Section list data should have data array.

**Comment by devkudasov** [comment]
Close as old and not relevant for now

**Pull request closed without merging**